systemvm/patches/debian/config/opt/cloud/bin/configure.py

```diff
@@ -145,40 +145,41 @@ def add_rule(self, cidr):
             logging.debug("Current ACL IP direction is ==> %s", self.direction)
             if self.direction == 'egress':
                 self.fw.append(["filter", "", " -A FW_OUTBOUND -j FW_EGRESS_RULES"])
+                fwr = " -I FW_EGRESS_RULES"
+
+                #In case we have a default rule (accept all or drop all), we have to evaluate the action again.
+                if self.rule['cidr'] == ['0.0.0.0/0']:
```

Comment:
If-else block does exactly that logic. If rule has CIDR 0.0.0.0/0 then rule action is set to that of default egress policy in the network offering. If rule has a CIDR other than 0.0.0.0/0 then reverse action (as that of default egress policy in the network offering) is set for the rule